emu_base/base_classes/config.py

```diff
@@ -58,6 +58,27 @@ def __init__(
                 "Interaction matrix must be provided as a Python list of lists of floats"
             )
 
+        def is_symmetric_zero_diag_matrix(
```

Comment:
you are defining this as a _nested function_? There is no reason to do so
also I don't like that you can pass None, keeping the None logic outside of this function is best I think

Reply:
I removed the function. Now it should be way better